homeassistant/components/minecraft_server/coordinator.py

```diff
@@ -0,0 +1,94 @@
+"""The Minecraft Server integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+from typing import Any
+
+from mcstatus.server import JavaServer
+
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from . import helpers
+from .const import SCAN_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class MinecraftServerData:
+    """Representation of Minecraft Server data."""
+
+    latency: float
+    motd: str
+    players_max: int
+    players_online: int
+    players_list: list[str]
+    protocol_version: int
+    version: str
+
+
+class MinecraftServerCoordinator(DataUpdateCoordinator[MinecraftServerData]):
+    """Minecraft Server data update coordinator."""
+
+    def __init__(
+        self, hass: HomeAssistant, unique_id: str, config_data: Mapping[str, Any]
+    ) -> None:
+        """Initialize coordinator instance."""
+        super().__init__(
+            hass=hass,
+            name=config_data[CONF_NAME],
+            logger=_LOGGER,
+            update_interval=timedelta(seconds=SCAN_INTERVAL),
+        )
+
+        # Server data
+        self.unique_id = unique_id
```

Comment:
For what do we need the unique_id?

Reply:
The unique ID is actually the config entry ID and it's used by the platform entitites to set their entity unique IDs:

Example MinecraftServerSensorEntity in sensor.py:
`self._attr_unique_id = f"{coordinator.unique_id}-{description.key}"`

Reply:
Somehow my IDE didn't showed any usage of it.